src/libraries/icubmod/embObjMotionControl/eomcParser.h

```diff
@@ -56,6 +57,181 @@ namespace yarp {
 //    PidAlgo_currentInnerLoop =2
 //} PidAlgorithmType_t;
 
+class ITemperatureSensor
+{
+
+public:
+    // virtual ~ITemperatureSensor() = default; //TODO: note -> if I use smart pointer I do not need the destructor - theoretically, since it gets deleted when it goes out of scope - 
+
+    virtual void setTemperatureSensor() = 0;
+    virtual double convertTempCelsiusToRaw(const double temperature) = 0;
```

Comment:
Both `ConvertTempCelsiusToRaw` and `convertRawToTempCelsius` should have the same signature. I suggest returning the converted value.

You could add the sensor type and a method to get it in order to call the conversion functions only on the object different from NONE.